src/masonite/tests/HttpTestResponse.py

```diff
@@ -37,8 +52,221 @@ def assertIsNotNamed(self, name=None):
+        """Assert that session contains the given key with the corresponding value if given.
+        The session driver can be specified if necessary."""
+        session = self.request.session.driver(driver)
+        session_value = session.get(key)
+        assert session_value
+        if value is not None:
+            assert session_value == value
+        return self
+
+    def assertSessionMissing(self, key, driver="cookie"):
+        """Assert that session does not contain the given key. The session driver can be specified
+        if necessary."""
+        assert not self.request.session.driver(driver).get(key)
+        return self
+
+    def _ensure_response_has_view(self):
+        """Ensure that the response has a view as its original content."""
+        if not (self.response.original and isinstance(self.response.original, View)):
+            raise ValueError("The response is not a view")
+
+    def assertViewIs(self, name):
+        """Assert that request renders the given view name."""
+        self._ensure_response_has_view()
+        assert self.response.original.template == name
+        return self
+
+    def assertViewHas(self, key, value=None):
+        """Assert that view context contains a given data key (and eventually associated value)."""
+        self._ensure_response_has_view()
+        assert key in self.response.original.dictionary
+        if value:
+            assert self.response.original.dictionary[key] == value
+        return self
+
+    def assertViewHasAll(self, keys):
+        """Assert that view context contains exactly the data keys (or the complete data dict)."""
+        self._ensure_response_has_view()
+        if isinstance(keys, list):
+            assert set(keys) == set(self.response.original.dictionary.keys()) - set(self.response.original._shared.keys())
+        else:
+            view_data = self.response.original.dictionary
+            for key in self.response.original._shared:
+                del view_data[key]
+            assert keys == view_data
+        return self
+
+    def assertViewMissing(self, key):
+        """Assert that given data key is not in the view context."""
+        self._ensure_response_has_view()
+        assert key not in self.response.original.dictionary
+        return self
+
+    def assertAuthenticated(self):
+        assert self.application.make("auth").guard("web").user()
+        return self
+
+    def assertGuest(self):
+        assert not self.application.make("auth").guard("web").user()
+        return self
+
+    def assertAuthenticatedAs(self, user):
+        user = self.application.make("auth").guard("web").user()
+        assert user == user
+        return self
+
+    def assertHasHttpMiddleware(self, middleware):
+        """Assert that the request/response cycle has the given middleware. The HTTP middleware
+        class should be given."""
+        assert middleware in self.application.make("middleware").http_middleware
```

Comment:
@josephmancuso  not sure here if we should make one assertion method for all middlewares ?
Also middleware are registered with class in a list for HTTP and with a dict for routes one, so not sure what the user wants to check for now it would be
```python
assertHasHttpMiddleware(MyCustomMiddleware)
assertHasRouteMiddleware("web")
```
I could use * operator to be able to do assertHasRouteMiddleware("web", "auth") 🤔  ?

Reply:
Yeah thats a good idea. Can always convert both to tuples and to issubset to check for the intersections between the 2 tuples. List might also have this support 

https://www.programiz.com/python-programming/methods/set/issubset

Reply:
```python
    def assertHasHttpMiddleware(self, *middlewares):
        """Assert that the request/response cycle has the given middleware. The HTTP middleware
        class should be given."""
        assert set(middlewares).issubset(set(self.application.make("middleware").http_middleware)) 
```

something like this. Idk if it works